scripts/screeps-private-smoke.py

```
@@ -0,0 +1,1431 @@
+            "source": str(cfg.map_source_file) if cfg.map_source_file else cfg.map_url,
+            "status": "planned",
+        }
+    if cfg.map_source_file:
+        if not cfg.map_source_file.exists():
+            raise SmokeError(f"map source file does not exist: {cfg.map_source_file}")
+        shutil.copyfile(cfg.map_source_file, cfg.map_path)
+        source = str(cfg.map_source_file)
+    elif cfg.map_path.exists():
+        source = "existing-workdir-file"
+    else:
+        request = urllib.request.Request(cfg.map_url, headers={"User-Agent": "screeps-private-smoke/1.0"})
```

Ruff (0.15.11)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)